krate/src/main/kotlin/hu/autsoft/krate/ValidatedFunctions.kt

```diff
@@ -0,0 +1,116 @@
+ * an [IllegalArgumentException] will be thrown.
+ */
+public fun Krate.intPref(key: String, isValid: (Int?) -> Boolean): ReadWriteProperty<Krate, Int?> {
+    return ValidatedPreferenceDelegate(IntDelegate(key), isValid)
+}
+
+/**
+ * Creates a validated, non-optional preference of type [Int] with the given [key].
+ *
+ * If a value being set to this preference returns `false` when checked by [isValid],
+ * an [IllegalArgumentException] will be thrown.
+ */
+public fun Krate.intPref(key: String, default: Int, isValid: (Int) -> Boolean): ReadWriteProperty<Krate, Int> {
+    return ValidatedPreferenceDelegate(IntDelegateWithDefault(key, default), isValid)
+}
+
+/**
+ * Creates a validated, optional preference of type [Long] with the given [key].
+ *
+ * If a value being set to this preference returns `false` when checked by [isValid],
+ * an [IllegalArgumentException] will be thrown.
+ */
+public fun Krate.longPref(key: String, isValid: (Long?) -> Boolean): ReadWriteProperty<Krate, Long?> {
+    return ValidatedPreferenceDelegate(LongDelegate(key), isValid)
+}
+
+/**
+ * Creates a validated, non-optional preference of type [Long] with the given [key].
+ *
+ * If a value being set to this preference returns `false` when checked by [isValid],
+ * an [IllegalArgumentException] will be thrown.
+ */
+public fun Krate.longPref(key: String, defaultValue: Long, isValid: (Long) -> Boolean): ReadWriteProperty<Krate, Long> {
+    return ValidatedPreferenceDelegate(LongDelegateWithDefault(key, defaultValue), isValid)
+}
+
+/**
+ * Creates a validated, optional preference of type [String] with the given [key].
+ *
+ * If a value being set to this preference returns `false` when checked by [isValid],
+ * an [IllegalArgumentException] will be thrown.
+ */
+public fun Krate.stringPref(key: String, isValid: (String?) -> Boolean): ReadWriteProperty<Krate, String?> {
+    return ValidatedPreferenceDelegate(StringDelegate(key), isValid)
+}
+
+/**
+ * Creates a validated, non-optional preference of type [String] with the given [key].
+ *
+ * If a value being set to this preference returns `false` when checked by [isValid],
+ * an [IllegalArgumentException] will be thrown.
+ */
+public fun Krate.stringPref(key: String, default: String, isValid: (String) -> Boolean): ReadWriteProperty<Krate, String> {
+    return ValidatedPreferenceDelegate(StringDelegateWithDefault(key, default), isValid)
+}
+
+/**
+ * Creates a validated, optional preference of type [Set] with the given [key].
+ *
+ * If a value being set to this preference returns `false` when checked by [isValid],
+ * an [IllegalArgumentException] will be thrown.
+ */
+public fun Krate.stringSetPref(key: String, isValid: (Set<String>?) -> Boolean): ReadWriteProperty<Krate, Set<String>?> {
+    return ValidatedPreferenceDelegate(StringSetDelegate(key), isValid)
+}
+
+/**
+ * Creates a validated, non-optional preference of type [Set] with the given [key].
```

Comment:
This `[Set]` here is a bit vague, the original docs say `Set<String>` instead. This does come with the downside that the `[]` syntax doesn't work around it anymore for navigating to the type, but that's probably fine. If we remove them here, they should be removed from all the similar types in these comments for uniformity.

Reply:
updated to coincide with the docs for the non validated functions. Omitted `[]` and just used `Set<String>`